modules/misc/qt.nix

```diff
@@ -44,19 +44,68 @@ in {
           </variablelist>
         '';
       };
+
+      style = mkOption {
+        type = types.nullOr (types.enum [
+          "adwaita"
+          "adwaita-dark"
+          "cleanlooks"
+          "gtk2"
+          "motif"
+          "plastique"
+        ]);
+        default = null;
+        example = "adwaita";
+        relatedPackages = [ "adwaita-qt" [ "libsForQt5" "qtstyleplugins" ] ];
+        description = ''
+          Selects the style to use for Qt5 applications.</para>
+          <para>The options are
+          <variablelist>
+            <varlistentry>
+              <term><literal>adwaita</literal></term>
+              <term><literal>adwaita-dark</literal></term>
+              <listitem><para>Use Adwaita Qt style with
+                <link xlink:href="https://github.com/FedoraQt/adwaita-qt">adwaita</link>
+              </para></listitem>
+            </varlistentry>
+            <varlistentry>
+              <term><literal>cleanlooks</literal></term>
+              <term><literal>gtk2</literal></term>
+              <term><literal>motif</literal></term>
+              <term><literal>plastique</literal></term>
+              <listitem><para>Use styles from
+                <link xlink:href="https://github.com/qt/qtstyleplugins">qtstyleplugins</link>
+              </para></listitem>
+            </varlistentry>
+          </variablelist>
+        '';
+      };
     };
   };
 
   config = mkIf (cfg.enable && cfg.platformTheme != null) {
-    home.sessionVariables.QT_QPA_PLATFORMTHEME =
-      if cfg.platformTheme == "gnome" then "gnome" else "gtk2";
+    assertions = [{
+      assertion = (cfg.platformTheme == "gnome")
+        -> (builtins.elem cfg.style [ "adwaita" "adwaita-dark" ]);
+      message = ''
+        `qt.platformTheme` "gnome" must have `qt.style` set to either "adwaita" or "adwaita-dark".'';
+    }];
+
+    home.sessionVariables = (filterAttrs (n: v: v != null) {
+      QT_QPA_PLATFORMTHEME =
+        if cfg.platformTheme == "gnome" then "gnome" else "gtk2";
+      QT_STYLE_OVERRIDE = cfg.style;
+    });
```

Comment:
Can you add a comment that this logic is necessary because `home.sessionVariables` is of `types.attrs`.